src/helpers.js

```diff
@@ -162,35 +181,20 @@ function registerWith(Handlebars) {
         }
 
         if (typeof message === 'string') {
-            message = getMessageFormat(message, locales, formats);
+            message = getMessageFormat(isHtml ? message : escape(message), locales, formats);
+        }
+
+        if (hasSafeString) {
+          return new SafeString(String(message.format(hash)));
         }
 
         return message.format(hash);
     }
 
-    function formatHTMLMessage() {
-        /* jshint validthis:true */
-        var options = [].slice.call(arguments).pop(),
-            hash    = options.hash;
-
-        var key, value;
-
-        // Replace string properties in `options.hash` with HTML-escaped
-        // strings.
-        for (key in hash) {
-            if (hash.hasOwnProperty(key)) {
-                value = hash[key];
-
-                // Escape string value.
-                if (typeof value === 'string') {
-                    hash[key] = escape(value);
-                }
-            }
-        }
-
+    function formatHTMLMessage(message, options) {
         // Return a Handlebars `SafeString`. This first unwraps the result to
         // make sure it's not returning a double-wrapped `SafeString`.
-        return new SafeString(String(formatMessage.apply(this, arguments)));
+        return new SafeString(String(formatMessage.call(this, message, options, true)));
```

Comment:
I generally don't like boolean params, what if instead we cue the `isHTML` behavior on whether the message string itself is an `instanceof SafeString`?